### PR TITLE
sig-storage, enable to always run

### PR DESCRIPTION
**Pull request by brybacki**
The sig-storage lane should be actually run on all PRs so we can compare it's behavior to other lanes (with all tests). 
In the near future we want to remove sig-storage tests from main 1.19 lane and from periodic job. 

It takes about 30 minutes to finish on average. 

No failure (not counting single image fetch problem) since creating a periodic job with this lane: https://prow.apps.ovirt.org/job-history/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.19-sig-storage

Signed-off-by: Bartosz Rybacki <brybacki@redhat.com>

### CODE REVIEW

**Comment by fgimenez** [comment]
/hold

Looks good but let's wait until we have more resources available.

**Comment by aglitke** [comment]
@fgimenez We want to make progress on reducing the size of the e2e lanes and spreading out ownership of tests to subject matter experts.  When are you expecting more resources in order to merge this?  Would it ease the path to merging if there was an accompanying PR that removed these same tests from the e2e lanes?

**Comment by fgimenez** [comment]
> When are you expecting more resources in order to merge this?

We are building a new cluster to run e2e workloads, it has already attached 2 new bare-metal machines and we are running test jobs on it, I believe it will be ready during this week. As soon as it is available we can modify this PR to run the jobs in that cluster and merge it.

> Would it ease the path to merging if there was an accompanying PR that removed these same tests from the e2e lanes?

Absolutely, as I understand it that was the ultimate goal of splitting the big e2e lanes, we should definitely aim to remove the sig-related tests from there.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 1]
```diff
@@ -150,7 +150,7 @@ presubmits:
     annotations:
       fork-per-release: "true"
       testgrid-dashboards: kubevirt-presubmits
-    always_run: false
+    always_run: true
```
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:153]: Let's start running this job in the new cluster:
```suggestion
    always_run: true
    cluster: prow-workloads
```

**Comment by brybacki** [comment]
It should be done the same way as `sig-network` change https://github.com/kubevirt/project-infra/pull/1081

I'll update and push the changes later

**Comment by fgimenez** [comment]
/unhold

I think we can start testing this lane in the new cluster, you would need to:
* Include a `cluster: prow-workloads` entry in the job definition (change already proposed)
* turn off docker proxy by setting this label to false https://github.com/kubevirt/project-infra/blob/e59e3294632cba12c1e08567f67ad7e02a9e9c40/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L163
* turn off bazel cache by setting these two labels to false: https://github.com/kubevirt/project-infra/blob/e59e3294632cba12c1e08567f67ad7e02a9e9c40/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L165 and https://github.com/kubevirt/project-infra/blob/e59e3294632cba12c1e08567f67ad7e02a9e9c40/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L166

In order to make /rehearse work you would need to modify the job spec, maybe adding something like `&& echo 'Changes for rehearse'` at the end of https://github.com/kubevirt/project-infra/blob/e59e3294632cba12c1e08567f67ad7e02a9e9c40/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L176

**Comment by brybacki** [comment]
looks good, let me try again
/rehearse

**Comment by fgimenez** [comment]
great, we might need more changes in the job spec to trigger rehearse again

**Comment by brybacki** [comment]
/rehearse

right, some dummy change needed to be able to rehearse

**Comment by brybacki** [comment]
@fgimenez  two questions:
1. since we have a new cluster configured for sig-storage lane, maybe we can go ahead with the change with two first commits:
```
sig-storage: run on prow-workloads cluster …
sig-storage: made mandatory and always run …
```
and the remaining two (to skip sig-storage on 1.19 lanes) do in another PR?

2. I have trouble in deterministically doing a rehearse, I did try adding echo to a command in a few places, and I managed to run sig-storage (with success) but the 1.19 did not run.

**Comment by fgimenez** [comment]
Sorry for the late response,

> 1. since we have a new cluster configured for sig-storage lane, maybe we can go ahead with the change with two first commits:
> 
> ```
> sig-storage: run on prow-workloads cluster …
> sig-storage: made mandatory and always run …
> ```
> 
> and the remaining two (to skip sig-storage on 1.19 lanes) do in another PR?

ok, sounds good to me :+1: 



> 1. I have trouble in deterministically doing a rehearse, I did try adding echo to a command in a few places, and I managed to run sig-storage (with success) but the 1.19 did not run.

AIUI it only triggers jobs that have changes in the spec section, did you change 1.19's spec?

**Comment by brybacki** [comment]
well, now I need to rebase, to get sig-network changes

**Comment by brybacki** [comment]
rebased

**Review by fgimenez — APPROVED** [review]
/lgtm
/approve

Thanks!
